andela/octopus-bn-backend
# #170947429 setup swagger

#### What does this PR do?
setup api documentation using swagger

#### Description of Task to be completed?
- setup swagger for auto-documentation using jsdoc

#### How should this be manually tested?
- Use `git clone <repo link>` to clone the repo to your local machine
- Use `npm i` to install all the project dependencies
- Run `npm run dev` to start the development server
- Access the documentation on this Link ```http://localhost:3000/api-docs/```
- You can also check the hosted link of this documentation by clicking on `show environments` right next to `This branch was successfully deployed`, then click on `view deployment` and add this `/api-docs` on the url.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#170947429](https://www.pivotaltracker.com/story/show/170947429)

#### Screenshots (if appropriate)
<img width="1195" alt="Screen Shot 2020-02-06 at 12 46 29" src="https://user-images.githubusercontent.com/53453380/73930146-bc75d580-48de-11ea-92d0-01c02921b703.png">

#### Questions:
N/A
